### DISCUSSION TITLE
Print quality on zebra printers

### DISCUSSION THREAD

**Discussion by Pixeltrace-GER** [inventree/InvenTree#4494] [Q&A] [answered]
Hi Guys,
I have a small print quality issue. But at first, some sentence about my setup.
Inventree is running on a V-Server (Docker/Production). I generate the Label as PDF to pint them on my local system.
One week ago, I switch the Printer from Dymo to Zebra. Since this change, I'm not happy with my print quality.

I have done some testing in this setup.
- Printing Labels over ZDesigner (Label software by Zebra) -> no issues.
- Printing PDF Labels (generate at Labelary.com) -> no issues.
- Printing Inventree Label as PDF -> poor quality

Can someone point me in the right direction, what I should check next?
The PDF generation is set to 600dpi in the settings.

[Inventree_Labelary.png](https://github.com/Pixeltrace-GER/inventree-label-issues/blob/master/Inventree_Labelary.png)

**Comment by matmair** [comment]
@SergeoLacruz might have some insights here, he also uses Zebra printers.

**Comment by wolflu05** [comment]
@Pixeltrace-GER One quick off topic question, how have you connected your dymo printer to inventree, and what dymo printer have you used?

**Reply by Pixeltrace-GER** [reply]
As I mentioned in my first post, Inventree is running on a V-Server. So, there is no easy way to connect a Printer to the Server.
My old printer was Dymo LW450. Connected to the Local PC, printing the PDF Label as a normal Windows printer.

**Reply by wolflu05** [reply]
I actually have the same model, except I have the duo one. If you still using this, I developed a [cups plugin](https://github.com/wolflu05/inventree-cups-plugin). You can utilize a raspberry pi zero if you have one or any other device as a cups server and connect the printer to it. Then you can put your printer wherever you want and print over network.

**Reply by Pixeltrace-GER** [reply]
Yes, I have seen your plugin. But for now, I go with the Zebra

**Comment by SergeoLacruz** [comment]
Hi,
Might be a scaling issue. You have pdf using 600dpi. How do you convert this to zpl? 
Zebra printers use 203 dpi. During conversion from 600 to 203 you might end up with a grayscale picture. Zebras cannot print grayscale and convert this to BW. the algo in the Zebra doing that is trash. I used the pillow convert to do the grey to bw conversion and ended up with far better results. Have a look here:

https://github.com/SergeoLacruz/inventree-zebra-plugin/blob/master/inventree_zebra/zebra_plugin.py

I am satisfied with the result. Even 10x10mm QR is fine and can easily be read by Honeywell hand scanners or even an Iphone.
I use gk420 thermo transfer on plastic labels. 

Michael

**Reply by Pixeltrace-GER** [reply]
Hi Michael,
I don't convert it to ZPL. I just print the PDF. Converting is happening in the windows/zebra driver.
My printer is a 300dpi version. Maybe this is a Point. 

Since yesterday, I try to rewrite your Macro to download the ZPL. But I'm learning Django since on week :D

**Comment by SergeoLacruz** [comment]
I do not know what the driver does...
If you do not have the printer on the server you can use the network interface in my code. Set up a simple server on the computer where the printer is installed. There is an example also in the repo. I use this setup for development because I also have the printer elsewhere and it has only USB :-(

**Reply by Pixeltrace-GER** [reply]
Okay, thanks for your help. :)
I will test some more things. 

My printer has a Network card, so this would be a way to use your plugin. Bat then i need a VPN Connection from the Server to my Home network... this is my last option to get it running :D

If I have a solution, I will let you guys know my way.

**Comment by Pixeltrace-GER** [comment, marked as answer]
Hi everybody, 
I found a solution.
There is one setting in the Windows driver.

**Graphic options -> Dithering**
Move the slider fully to the right. So no dithering. Only Black and White. 

Now i'm happy again :)

[Zebra - support article](https://supportcommunity.zebra.com/s/article/Adjusting-image-quality-in-ZebraDesigner-3)
